src/main/java/blusunrize/immersiveengineering/common/blocks/wooden/LogicUnitTileEntity.java

```diff
@@ -0,0 +1,245 @@
+import net.minecraft.util.Direction;
+import net.minecraft.util.NonNullList;
+import net.minecraftforge.client.MinecraftForgeClient;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.util.LazyOptional;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class LogicUnitTileEntity extends IEBaseTileEntity implements ITickableTileEntity, IIEInventory,
+		IInteractionObjectIE, IStateBasedDirectional, ILogicCircuitHandler, IHasObjProperty
+{
+	private final NonNullList<ItemStack> inventory = NonNullList.withSize(10, ItemStack.EMPTY);
+
+	public LogicUnitTileEntity()
+	{
+		super(IETileTypes.LOGIC_UNIT.get());
+	}
+
+	@Override
+	public Property<Direction> getFacingProperty()
+	{
+		return IEProperties.FACING_HORIZONTAL;
+	}
+
+	@Override
+	public PlacementLimitation getFacingLimitation()
+	{
+		return PlacementLimitation.HORIZONTAL;
+	}
+
+	@Override
+	public boolean mirrorFacingOnPlacement(LivingEntity placer)
+	{
+		return placer.isSneaking();
+	}
+
+	@Override
+	public void tick()
+	{
+	}
+
+	@Override
+	public void readCustomNBT(CompoundNBT nbt, boolean descPacket)
+	{
+		ItemStackHelper.loadAllItems(nbt, inventory);
+	}
+
+	@Override
+	public void writeCustomNBT(CompoundNBT nbt, boolean descPacket)
+	{
+		ItemStackHelper.saveAllItems(nbt, inventory);
+	}
+
+	@Override
+	public boolean canUseGui(PlayerEntity player)
+	{
+		return true;
+	}
+
+	@Override
+	public IInteractionObjectIE getGuiMaster()
+	{
+		return this;
+	}
+
+	@Override
+	public NonNullList<ItemStack> getInventory()
+	{
+		return inventory;
+	}
+
+	@Override
+	public boolean isStackValid(int slot, ItemStack stack)
+	{
+		return stack.getItem() instanceof LogicCircuitBoardItem;
+	}
+
+	@Override
+	public int getSlotLimit(int slot)
+	{
+		return 1;
+	}
+
+	@Override
+	public void doGraphicalUpdates(int slot)
+	{
+		this.markDirty();
+		redstoneCap.ifPresent(RedstoneBundleConnection::markDirty);
+	}
+
+	private final static int SIZE_COLORS = DyeColor.values().length;
+	private final static int SIZE_REGISTERS = LogicCircuitRegister.values().length-SIZE_COLORS;
+	EnumMap<Direction, boolean[]> inputs = new EnumMap<>(Direction.class);
+	boolean[] registers = new boolean[SIZE_REGISTERS];
+	boolean[] outputs = new boolean[SIZE_COLORS];
+
+	private boolean runCircuits()
+	{
+		boolean[] outPre = Arrays.copyOf(outputs, SIZE_COLORS);
+		this.inventory.stream().map(LogicCircuitBoardItem::getInstruction).filter(Objects::nonNull)
+				.forEachOrdered(instruction -> {
+					instruction.apply(this);
+				});
+		return !Arrays.equals(outPre, outputs);
+	}
+
+	private void markConnectorsDirty()
+	{
+		for(Direction d : Direction.values())
+		{
+			TileEntity te = Utils.getExistingTileEntity(world, getPos().offset(d));
+			if(te instanceof ConnectorBundledTileEntity&&((ConnectorBundledTileEntity)te).getFacing()==d.getOpposite())
+				((ConnectorBundledTileEntity)te).markDirtyExtraSource();
```

Comment:
After actually looking at the code my suggestion would be to have one `RedstoneBundleConnection` instance for each side, and to use the existing `markDirty`/`pollDirty` flow. I assume you didn't use it because the first connector would clear the `dirty`-flag before the others can react to it?

Reply:
Yep, that's exactly why that didn't work.

Reply:
What about my "one `RedstoneBundleConnection` instance per side" suggestion? If each side has its own instance everything should work correctly as far as I can see.